contracts/CYFI.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity ^0.5.16;
+
+import "./CErc20.sol";
+
+interface IDelegateRegistry {
+    function setDelegate(bytes32 id, address delegate) public;
+}
+
+contract CYFI is CErc20 {
+
+    IDelegateRegistry public delegateRegistry;
+
+    /**
+     * @notice Construct a new money market
+     * @param underlying_ The address of the underlying asset
+     * @param comptroller_ The address of the Comptroller
+     * @param interestRateModel_ The address of the interest rate model
+     * @param initialExchangeRateMantissa_ The initial exchange rate, scaled by 1e18
+     * @param name_ ERC-20 name of this token
+     * @param symbol_ ERC-20 symbol of this token
+     * @param decimals_ ERC-20 decimal precision of this token
+     * @param admin_ Address of the administrator of this token
+     */
+    constructor(address underlying_,
+                ComptrollerInterface comptroller_,
+                InterestRateModel interestRateModel_,
+                uint initialExchangeRateMantissa_,
+                string memory name_,
+                string memory symbol_,
+                uint8 decimals_,
+                address payable admin_,
+                IDelegateRegistry delegateRegistry_) public {
+        // Creator of the contract is admin during initialization
+        admin = msg.sender;
+
+        // Initialize the market
+        initialize(underlying_, comptroller_, interestRateModel_, initialExchangeRateMantissa_, name_, symbol_, decimals_);
+
+        // Set the proper admin now that initialization is done
+        admin = admin_;
+
+        // set delegate registry
+        delegateRegistry = delegateRegistry_;
+    }
+
+    function delegate(address delegate_) public {
+        require(msg.sender == admin, "only admin may delegate");
+        delegateRegistry.setDelegate('ybaby.eth', delegate_);
```

Comment:
A comment to indicate that `ybaby.eth` is the id used for YFI's governance would be nice for future readers.